src/Pulse.php

```diff
@@ -508,15 +521,20 @@ public function handleExceptionsUsing(callable $callback): self
     /**
      * Execute the given callback handling any exceptions.
      *
-     * @param  (callable(): mixed)  $callback
+     * @template TReturn
+     *
+     * @param  (callable(): TReturn)  $callback
+     * @return TReturn|null
      */
-    public function rescue(callable $callback): void
+    public function rescue(callable $callback): mixed
     {
         try {
-            $callback();
+            return $callback();
         } catch (Throwable $e) {
             ($this->handleExceptionsUsing ?? fn () => null)($e);
         }
+
+        return null;
```

Comment:
I've checked all places where this could be impactful.

The only place it could be an issue is if someone returned `false` from a recorder. It could bail an `$collection->each()` call, but "don't hold it that way"